core/src/main/scala/org/apache/spark/status/api/v1/api.scala

```diff
@@ -109,7 +114,8 @@ class ExecutorSummary private[spark](
     @JsonDeserialize(using = classOf[ExecutorMetricsJsonDeserializer])
     val peakMemoryMetrics: Option[ExecutorMetrics],
     val attributes: Map[String, String],
-    val resources: Map[String, ResourceInformation])
+    val resources: Map[String, ResourceInformation],
+    val resourceProfileId: Int)
```

Comment:
While deserializing existing history files, which wont have resource profile related events, will it default to using ResourceProfile.DEFAULT_RESOURCE_PROFILE_ID ?
Or is it backwardly incompatible ?

Reply:
so the LiveEntity creates the ExecutorSummary and it has a default of the ResourceProfile.DEFAULT_RESOURCE_PROFILE_ID and executorInfoFromJson will default it to ResourceProfile.DEFAULT_RESOURCE_PROFILE_ID when that field isn't found as well.
So it will just default to that for older history files that don't have this field.

I had tested this with some 2.x history files originally and it was working that way as expected. But I should retest again after the changes from review comments but I don't think anything has broken that.  

Definitely something we want to make sure its backwards compatible.

Reply:
Thanks Tom ! My only concern was around what happens if the resource event is not fired (which it wont for older files) and how this works w.r.t existing jsons which dont have rp field populated.

Once you have validated this, rest looked good to me - but I am not really an expert on the UI side of things, so someone more familiar should probably take a look too.